pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -0,0 +1,343 @@
+package projectreference
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/google/uuid"
+	gcpv1alpha1 "github.com/openshift/gcp-project-operator/pkg/apis/gcp/v1alpha1"
+	"github.com/openshift/gcp-project-operator/pkg/gcpclient"
+	"github.com/openshift/gcp-project-operator/pkg/util"
+	operrors "github.com/openshift/gcp-project-operator/pkg/util/errors"
+	"google.golang.org/api/cloudresourcemanager/v1"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/iam/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	osdServiceAccountName = "osd-managed-admin"
+)
+
+// OSDRequiredAPIS is list of API's, required to setup
+// OpenShift cluster. Order is important.
+var OSDRequiredAPIS = []string{
+	"serviceusage.googleapis.com",
+	"cloudresourcemanager.googleapis.com",
+	"storage-component.googleapis.com",
+	"storage-api.googleapis.com",
+	"dns.googleapis.com",
+	"iam.googleapis.com",
+	"compute.googleapis.com",
+}
+
+var OSDRequiredRoles = []string{
```

Comment:
Linter-wise, since it starts with uppercase, it would be exported, and we need a comment/description.

```suggestion
// OSDRequiredRoles blah blah blah
var OSDRequiredRoles = []string{
```

Replace my _blahblahblah_ with the purpose its fits (I guess the reply to my question above, would fit for a description).

Reply:
Also, according to the [Required GCP Permissions](OSDRequiredRoles) there is a mismatch. In some case you use `iam.` while the docs are using `compute.`. Have a look at the link and see if this OK.